core/deployment/src/main/java/io/quarkus/deployment/CracConfig.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.deployment;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(phase = ConfigPhase.BUILD_TIME)
+public class CracConfig {
+
+    /**
+     * Enable/Disable CRAC registration
+     * Default value is dependent on extensions deployed
+     * (i.e. with Lambda this will be set to true by default)
+     */
+    @ConfigItem
+    Optional<Boolean> enable;
```

Comment:
We do need a better way of saying "the default of this build time value depends on programmatic factor X" than `Optional<Boolean>` which always reminds me of [this classic gem](https://thedailywtf.com/articles/what_is_truth_0x3f_). We do have `io.quarkus.deployment.builditem.RunTimeConfigurationDefaultBuildItem` but obviously this only reaches to run time... but that's a problem for later I guess...